### PR TITLE
SWARM-1442: Include licenses deliverables.

### PR DESCRIPTION
**Pull request by mkouba**
**NOT READY FOR MERGE YET!**

This PR outlines additional configuration needed to generate licenses deliverables using `wildfly-swarm-fraction-plugin` (currently unreleased 61-SNAPSHOT).

To try it out run:
> mvn clean package -DskipTests -Pgenerate-licenses
> mvn clean package -DskipTests -Pgenerate-licenses -f pom-redhat.xml

The generated zip can be found at `target/wfswarm-rest-http-8-SNAPSHOT-license.zip`.

Note that the `licenses-transform.xsl` file will need some more tweaks as it looks like there are some artifacts which don't have license specified etc.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -162,5 +167,89 @@
         </plugins>
       </build>
     </profile>
+
+      <!-- Activate this profile to generate licenses deliverables -->
+      <!-- Note that we cannot use id "profile" as it's already defined by booster parent -->
```
Comment (kenfinnigan) [inline pom.xml]: Should that be "licenses" and not "profile"?

File: src/main/resources/licenses/custom-settings.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+   xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
+                      http://maven.apache.org/xsd/settings-1.0.0.xsd">
+
+   <profiles>
+      <profile>
+         <id>local-m2</id>
```
Comment (kenfinnigan) [inline src/main/resources/licenses/custom-settings.xml:8]: Do we need an equivalent version of this file for generating licenses.xml for product?

Can that be easily done in a single branch?

File: pom.xml (hunk 1 above)
Comment (mkouba) [reply pom.xml]: Ah, yes of course. It should be _"licenses" profile id_.

File: src/main/resources/licenses/custom-settings.xml (hunk 2 above)
Comment (mkouba) [reply src/main/resources/licenses/custom-settings.xml:8]: Good point. I think we can use a property for jboss repo url and add one more profile with product value.

Comment (mkouba) [reply src/main/resources/licenses/custom-settings.xml:8]: @kenfinnigan Is it safe to share the product repo URL publicly?

Comment (kenfinnigan) [reply src/main/resources/licenses/custom-settings.xml:8]: We have it present in the pom-redhat.xml so I imagine we're ok

Comment (mkouba) [reply src/main/resources/licenses/custom-settings.xml:8]: Ok. I forgot about the `pom-redhat.xml` - we should copy the new profile there as well.

**Review by kenfinnigan — APPROVED** [review]
Think it looks good Martin.

If you're happy with what it's producing, lets roll it out and we can refine from there

**Comment by mkouba** [comment]
@kenfinnigan The PR was merged with a SNAPSHOT dependency - `wildfly-swarm-fraction-plugin`. I'm preparing another PR to sync with https://github.com/wildfly-swarm/wildfly-swarm/pull/638. So I can update the version once 61 is released.

**Pull request merged**